src/core/authentication/authentication.service.ts

```diff
@@ -1,20 +1,44 @@
+import { AxiosError } from 'axios';
+import { HttpService } from '@nestjs/axios';
 import { ConfigService } from '@nestjs/config';
 import { Inject, Injectable, LoggerService } from '@nestjs/common';
+import { Session } from '@ory/kratos-client';
 import { ConfigurationTypes, LogContext } from '@common/enums';
 import { WINSTON_MODULE_NEST_PROVIDER } from 'nest-winston';
 import { UserService } from '@domain/community/user/user.service';
 import { AgentInfo } from './agent-info';
 import { OryDefaultIdentitySchema } from './ory.default.identity.schema';
 import { NotSupportedException } from '@common/exceptions';
 import { AgentService } from '@domain/agent/agent/agent.service';
+import { getBearerToken } from './get.bearer.token';
+
 @Injectable()
 export class AuthenticationService {
+  private readonly kratosPublicUrl: string;
+  private readonly kratosPrivateUrl: string;
+  private readonly adminPasswordIdentifier: string;
+  private readonly adminPassword: string;
+  private readonly extendAfter: number;
+
   constructor(
     private configService: ConfigService,
     private agentService: AgentService,
     private userService: UserService,
+    private httpService: HttpService,
     @Inject(WINSTON_MODULE_NEST_PROVIDER) private readonly logger: LoggerService
-  ) {}
+  ) {
+    this.kratosPublicUrl = this.configService.get(
+      ConfigurationTypes.IDENTITY
+    ).authentication.providers.ory.kratos_public_base_url_server;
+    this.extendAfter = Number(
+      this.configService.get(ConfigurationTypes.IDENTITY).authentication
+        .providers.ory.earliest_possible_extend
+    );
+    // todo: read from env
+    this.kratosPrivateUrl = '';
+    this.adminPasswordIdentifier = '';
```

Comment:
Add the `adminPasswordIdentifier` and `adminPassword` in the config, below earliest_possible_extend:

```yml
        earliest_possible_extend: ${AUTH_EARLIEST_POSSIBLE_SESSION_EXTEND}:86400000

        admin_service_account:
          username: ${KRATOS_ADMIN_USERNAME}:kratos@alkem.io
          password: ${KRATOS_ADMIN_PASSWORD}:kr@t0$!
```